frontend/src/abilities.ts

```diff
@@ -19,7 +19,7 @@ class Data {
   }
 }
 
-export class SQRForm extends Data {
+export class SQRFormAbilityData extends Data {
```

Comment:
Why do we ever need to define this class? Can't we just pass the data in?

Reply:
With default behavior the data passed in needs to have a constructor. https://stalniy.github.io/casl/abilities/2017/07/21/check-abilities.html#instance-checks